### PR TITLE
v100.0.0

### PR DESCRIPTION
**Pull request by codeman7**
# 100.0.0

In this major release we made improvements to Ink. As well as improvements in Dialogs,
Flexible Header, Snackbar, and TextFields.

## Breaking changes

The `inkColor` in `MDCInkView` is now `null_resettable`.

### Dialogs

We added the ability to customize MDCAlertController presentation animation.

```swift
let alertController = MDCAlertController()
alertController.presentationOpacityAnimationDuration = 0.5
alertController.presentationScaleAnimationDuration = 0.25
alertController.presentationInitialScaleFactor = 0.7
```

```objc
MDCAlertController *alertController = [[MDCAlertController alloc] init];
alertController.presentationOpacityAnimationDuration = 0.5;
alertController.presentationScaleAnimationDuration = 0.25;
alertController.presentationInitialScaleFactor = 0.7;
```

### FlexibleHeader

We added the ability to infer top safe are insets from a view controller.

```swift
let flexibleHeader = MDCFlexibleHeaderViewController()
flexibleHeader.inferTopSafeAreaInsetFromViewController = true
```

```objc
MDCFlexibleHeaderViewController *flexibleHeader = [[MDCFlexibleHeaderViewController alloc] init];
flexibleHeader.inferTopSafeAreaInsetFromViewController = YES;
```

### Dialogs

*new* property: `presentationOpacityAnimationDuration` in `MDCAlertController`.

*new* property: `presentationScaleAnimationDuration` in `MDCAlertController`.

*new* property: `presentationInitialScaleFactor` in `MDCAlertController`.

### FlexibleHeader

*new* property: `inferTopSafeAreaInsetFromViewController` in `MDCFlexibleHeaderViewController`.

### Ink

*modified* property: `inkColor` in `MDCInkView` to be `null_resettable`.

### ActionSheet

* [Standardize the examples. (#9334)](https://github.com/material-components/material-components-ios/commit/315140a146b3b8f6a2ee7b1d2bd1d664e4ebc038) (featherless)

### Banner

* [Update README.md (#9292)](https://github.com/material-components/material-components-ios/commit/20b32f415f7d8a22c873a87fc22af37d93dca71f) (Radek)

### Dialogs

* [Add `const` keyword to CGFloat and UIEdgeInsets constants in MDCDialogPresentationController. (#9325)](https://github.com/material-components/material-components-ios/commit/72e1aa88a0b5fe65118996d3defccbad2c30639d) (Randall Li)
* [Expose MDCAlertController presentation animation properties (#9314)](https://github.com/material-components/material-components-ios/commit/f08507474ceed7b48e1a1bb0f7fb1611bc99e9cf) (Bryan Oltman)

### FlexibleHeader

* [Add new flag to allow use of inferTopSafeAreaInsetFromViewController with topLayoutGuideAdjustmentEnabled (#9323)](https://github.com/material-components/material-components-ios/commit/194d08f305921dc087a4c3303324fa696e086295) (featherless)
* [Clean up unsupported situation when inferring the top safe area inset (#9317)](https://github.com/material-components/material-components-ios/commit/36981ec75c4b146a46a12314c2872a4479d50700) (featherless)

### Ink

* [Update ink color to be null resettable. (#9253)](https://github.com/material-components/material-components-ios/commit/60e467f65408536e5f6f9bdd5f8c071b03de558e) (Yarden Eitan)

### Snackbar

* [Add snapshot test showing reported shadow bug (#9309)](https://github.com/material-components/material-components-ios/commit/d7a242e07fa2498a5fb9859dacd4c58a765421d3) (Yarden Eitan)
* [Resolve truncation of shadow in Snackbars (#9315)](https://github.com/material-components/material-components-ios/commit/c5f7118b13db0331ec16a596b0763130af4eb74e) (Yarden Eitan)

### Tabs

* [Remove out of date color theming docs (#9330)](https://github.com/material-components/material-components-ios/commit/25ee86e3fc4c6c2a0c1e3c14517360c5a8e55961) (Andrew Overton)

### TextControls

* [Add swift storyboard example (#9328)](https://github.com/material-components/material-components-ios/commit/95889abf03927e75e7c6976d19c5101e1e8e3c9f) (Andrew Overton)

### TextFields

* [Ensures that constraints used for the trailing view are properly removed when a new trailing view is set (#9336)](https://github.com/material-components/material-components-ios/commit/f4ec87ac392175df9e2b6827cb5eccc1738519b2) (Andrew Overton)
* [Removes requirement that the textView delegate be set when informing layout delegate of a size change (#9337)](https://github.com/material-components/material-components-ios/commit/f8d9897aac396090e15140aed1800b4431f04684) (Andrew Overton)

## Multi-component changes

* [Use prefix on category name for UIApplication+AppExtensions so that category names don't conflict in ObjC namespace. (#9277)](https://github.com/material-components/material-components-ios/commit/b86e38f249cb28211def17da69add34fb50badf6) (Randall Li)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,113 @@
...
+
+```swift
+let flexibleHeader = MDCFlexibleHeaderViewController()
+flexibleHeader.inferTopSafeAreaInsetFromViewController = true
+```
+
+```objc
+MDCFlexibleHeaderViewController *flexibleHeader = [[MDCFlexibleHeaderViewController alloc] init];
+flexibleHeader.inferTopSafeAreaInsetFromViewController = YES;
+```
```
Comment (jverkoey) [inline CHANGELOG.md:43]: `permitInferringTopSafeAreaFromTopLayoutGuideViewController` is the new API.

```
When disabled, if both inferTopSafeAreaInsetFromViewController and topLayoutGuideAdjustmentEnabled are set to YES, and the view controller selected to extract the safe area inset from (either automatically or via the delegate) is the same as topLayoutGuideViewController, the app will crash.

When enabled, the app will not crash in the situation described above. This is only supported on iOS 11+.

Enable this property before setting inferTopSafeAreaInsetFromViewController or topLayoutGuideViewController.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,113 @@
+# 100.0.0
+
```
Comment (yarneo) [inline CHANGELOG.md:2]: would you be able to elaborate a bit what these improvements are? i.e. customization ability to Dialog’s animation

File: CHANGELOG.md (hunk 1 above)
Comment (codeman7) [reply CHANGELOG.md:43]: Addressed.

File: CHANGELOG.md (hunk 2 above)
Comment (codeman7) [reply CHANGELOG.md:2]: Addressed.

**Review by yarneo — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,117 @@
...
+In this major release we made improvements to Ink. As well as improvements including customization Dialogs presentation animation,
+Flexible Header behavior around safe area. Also, we addressed bugs in Snackbar and TextFields.
```
Comment (yarneo) [inline CHANGELOG.md:4]: tip: In this major release we made a breaking change improvement to Ink’s inkColor API.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,117 @@
...
+```objc
+MDCAlertController *alertController = [[MDCAlertController alloc] init];
+alertController.presentationOpacityAnimationDuration = 0.5;
+alertController.presentationScaleAnimationDuration = 0.25;
+alertController.presentationInitialScaleFactor = 0.7;
+```
+
+### FlexibleHeader
+
+When disabled, if both inferTopSafeAreaInsetFromViewController and topLayoutGuideAdjustmentEnabled are set to YES, and the view controller selected to extract the safe area inset from (either automatically or via the delegate) is the same as topLayoutGuideViewController, the app will crash.
```
Comment (jverkoey) [inline CHANGELOG.md:33]: Call out that you're talking about `permitInferringTopSafeAreaFromTopLayoutGuideViewController` above here.

> We added a new runtime flag `permitInferringTopSafeAreaFromTopLayoutGuideViewController` to address a crash that was occurring in certain configurations.

**Review by jverkoey — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (codeman7) [reply CHANGELOG.md:4]: Done

File: CHANGELOG.md (hunk 4 above)
Comment (codeman7) [reply CHANGELOG.md:33]: Done

**Pull request merged**